src/webpack/index.ts

```diff
@@ -75,3 +84,55 @@ function removeMiniCssExtract(config: any) {
     }
   });
 }
+
+function getLibraryName(options: any) {
+  if (options?.customWebpackConfig?.libraryName) {
+    return options.customWebpackConfig.libraryName;
+  }
+
+  const projectName = getProjectNameFromAngularJson();
+  if (projectName) return projectName;
+
+  console.warn(
+    'Warning: single-spa-angular could not determine a library name to use and has used a default value.',
+  );
+  console.info('This may cause issues if this app uses code-splitting or lazy loading.');
+  if (!options) {
+    console.info('You may also need to update extra-webpack.config.json.');
+  }
+  console.info('See <SomeUrl> for information on how to resolve this.');
+
+  return 'angular_single_spa_project';
```

Comment:
I wanted something better than 'app3' as the default. I'm certainly willing to accept suggestions for a better default.